Marlin/src/core/serial.h

```diff
@@ -62,7 +62,7 @@ extern uint8_t marlin_debug_flags;
 //
 // Serial redirection
 //
-#define SERIAL_ALL 0x7F
+#define SERIAL_ALL 0xFF
```

Comment:
Hum... The default as `0x7F` seems not a mistake... it appears to be intentional ... @X-Ryl669?

Reply:
It was only renamed in his serial refactor. It used to represent a non-negative but unlikely serial INDEX. Now it represents a MASK of up to 8 serial ports.

Reply:
That's because the serial index is signed and negative values have a special meaning (don't remember why right now, but they are few places where -1 is used for the serial index default value). So it's set to  7 bits (= 7 serial ports) and not 8 bits here.

Reply:
If you want to set it to 0xFF, then you'll need to change to `typedef uint8_t serial_index_t`, IMHO.

Reply:
`SERIAL_ALL` is not a serial index. It has always been a serial mask (previously could be considered as having 2 bits) and in examining the code I see no instance where is it used in the context of a serial index. So I think we're all good.

Reply:
Yep, but it's saved into a serial_index_t in the `PORT_REDIRECT` macro. And since the value `-1` is used as a default argument (in queue.cpp), setting `SERIAL_ALL` to 255 (which is -1), means that default value for the queue's functions now means all serials.

I'm not fond of mixing index and mask here. Either we should move to mask everywhere, or remove ambiguities (the worst that could happen is doing a `BV(serial_index)` when `serial_index` is -1).

Reply:
See queue.h line 82 and 88. I think it's mixing index and mask here. It should read `serial_index_t p = 0` instead of `-1` since that would redirect to the first serial port when not in multiserial case. And when this is done, serial_index_t could be changed to an unsigned value (makes more sense to me, IMHO).